src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java

```diff
@@ -85,14 +86,14 @@ public void finishBuildTrace(final BuildData buildData, final Run<?,?> run) {
         final String prefix = BuildPipelineNode.NodeType.PIPELINE.getTagName();
         final long endTimeMicros = buildData.getEndTime(0L) * 1000;
         buildSpan.setTag(DDTags.SERVICE_NAME, "jenkins");
-        buildSpan.setTag(DDTags.RESOURCE_NAME, buildData.getJobName(null));
+        buildSpan.setTag(DDTags.RESOURCE_NAME, removeBranchSuffix(buildData.getJobName(null)));
```

Comment:
For context, what is this tag for, why does the job name contains the branch in it, and are we sure this method will not remove important information?

Reply:
When the user triggers a multibranch pipeline, the job name contains a suffix with the branch. This suffix needs to be removed to treat all pipeline executions related to the same pipeline item as the same resources in Datadog (and not per branch)

Reply:
The issue with Jenkins though is that plugins have a lof of freedom. The multibranch plugin will create one "Project/Job" per branch in the repo and indeed will add the name of the branch to the Job name.
We're here tackling an specific behavior of one specific plugin. But other plugins can do similar things that won't be handled by this change.

I think a best bet would be limit this specific behavior to cases where we are sure that the Job comes from a multibranch pipeline. I'm sure there's a way to tell that without depending on the multibranch plugin either. Maybe check the list of action and check if one of the action's class name matches a multibranch action?

Reply:
This function will simply remove anything after the first `/`.
If a non multibranch project has a `/` in the name, the suffix will get removed unintentionally

Reply:
AFAIK, the `/` character is not allowed to be used as a name as a Jenkins item, so I'd not expect that the user has typed this character intentionally.  Taking this into account, if this character appears in the Jenkins job name implies that Jenkins itself has added some suffix that needs to be removed for the `resource.name` tag.

Reply:
I'm going to remove this specific logic from this PR. As you mention, more things need to be considered and it will be adjusted in other specific PR.